subsystems/indexersubsystem.py

```diff
@@ -0,0 +1,83 @@
+from enum import Enum, auto
+from commands2 import SubsystemBase
+from wpilib import SmartDashboard
+import constants
+from util.helpfulIO import Falcon, LimitSwitch
+
+
+class IndexerSubsystem(SubsystemBase):
+    class Mode(Enum):
+        Holding = auto()
+        FeedForward = auto()
+        Reversed = auto()
+        Off = auto()
+
+        def asString(self) -> str:
+            mapping = {
+                self.Holding: "Holding",
+                self.FeedForward: "Feed Forward",
+                self.Reversed: "Reversed",
+                self.Off: "Off",
+            }
+            return mapping[self]
+
+    def __init__(self) -> None:
+        SubsystemBase.__init__(self)
+        self.setName(__class__.__name__)
+        self.indexerMotor = Falcon(
+            constants.kIndexerMotorName,
+            constants.kIndexerMotorId,
+            constants.kSimIndexerMotorPort,
+        )
+        self.stagingMotor = Falcon(
+            constants.kStagingMotorName,
+            constants.kStagingMotorId,
+            constants.kSimStagingMotorPort,
+        )
+        self.indexerSensor = LimitSwitch(
+            self.indexerMotor,
+            constants.kForwardSensorIndexer,
+            constants.kSimIndexerSensorId,
+        )
+        self.stagingSensor = LimitSwitch(
+            self.stagingMotor,
+            constants.kForwardSensorStaging,
+            constants.kSimStagingSensorId,
+        )
+        self.state = self.Mode.Holding
+
+    def periodic(self) -> None:
+        SmartDashboard.putString(
+            constants.kIndexerSystemStateKey, self.state.asString()
+        )
+
+    # Switches direction to reverse the ball path
+    def motorsOff(self) -> None:
+        self.state = self.Mode.Off
+
+    def feedBallForward(self) -> None:
+        self.state = self.Mode.FeedForward
+
+    def holdBall(self) -> None:
+        self.state = self.Mode.Holding
+
+    def reversePath(self) -> None:
+        self.state = self.Mode.Reversed
+
+    def defaultIndexer(self) -> None:
+        if self.state == self.Mode.FeedForward:
+            self.indexerMotor.setSpeed(constants.kIndexerSpeed)
+            self.stagingMotor.setSpeed(constants.kStagingSpeed)
+        elif self.state == self.Mode.Holding:
+            if self.indexerSensor.value() and self.stagingSensor.value():
```

Comment:
Technically this only has to be the indexer sensor because the staging sensor will always be covered when the indexer one is; however, this would not account for the possibility of a passing ball activating it and therefore shutting off the motors, so let's stick with this.

TLDR: No change